### DISCUSSION TITLE
The RDP connection does not permit to two clients to connect at the same time

### DISCUSSION THREAD

**Discussion by civier** [neurodesk/neurodesk.github.io#117] [Q&A] [answered]
In Neurodesktop, the RDP connection (via the web browser) does not permit to two clients to connect at the same time, whereas the VNC connection (via web browser, that existed in older versions) does allow it. Is it a limitation of RDP, or are multiple connections can be enabled with some effort?
If RDP does not support multiple clients at the moment, can we have the VNC option back even at the price of slower connectivity? (we were using the multiple client options to support out users, and it's quite useful)

**Comment by stebo85** [comment, marked as answer]
Yes, I don’t think that RDP supports connections from multiple clients. You can enable VNC using the —vnc command line flag when starting the docker container. @aswinnarayanan is this flag documented somewhere?

**Reply by aswinnarayanan** [reply]
@civier
This is an RDP limitation. 

We've turned off VNC by default to reduce overheads.
To disable RDP and enable VNC,  use the `-e VNC_ENABLE=true` flag when running the docker command.
I've added the instructions to the respective docs
https://neurodesk.github.io/docs/neurodesktop/getting-started/cloud/#using-vnc

Let me know if there's any issues.

**Comment by civier** [comment]
Interesting. I might look into this, as I'm interested if this is really a limitation of the RDP protocol per-se or just an implementation issue.

Thanks for documenting the VNC option, very useful. I just wonder -- is there any reason why the flag both enables VNC and disables RDP? (why can't I have both, like before?). I can see a scenario where users by default work on the RDP connection (as it much faster), and that I use the VNC connection to connect to their neurodesktop only when in need to diagnose or check things.

Re the overheads introduced by VNC: is it only starting time or resources, or is it making the neurodesktop slower in general even if resources are abundant?

**Reply by aswinnarayanan** [reply]
Hi @civier. The vnc rdp flag functionality has been updated, it should now be able to do what you're asking. Documentation has been updated with the new flags https://neurodesk.github.io/docs/neurodesktop/getting-started/cloud/#using-vnc
